src/coordination/coordinator_instance.cpp

```diff
@@ -199,10 +320,17 @@ auto CoordinatorInstance::TryFailover() -> void {
 
   auto *new_main = &FindReplicationInstance(most_up_to_date_instance);
 
-  if (!raft_state_.AppendOpenLockFailover(most_up_to_date_instance)) {
+  if (!raft_state_.AppendOpenLock()) {
     spdlog::error("Aborting failover as instance is not anymore leader.");
     return;
   }
+
+  utils::OnScopeExit do_reset{[this]() {
```

Comment:
Merge with other OnScopeExit

Reply:
I think we don't need an on-scope exit to resume frequent checks anymore so I will delete it. Added comment to explain why I think we don't need it anymore

Reply:
It may not be primary use, but it can be used that way. I can refactor code, but let's sync first